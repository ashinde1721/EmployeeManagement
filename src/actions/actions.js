export const SEARCH_TEXT = "SEARCH_TEXT";
export const SAVE_EMPLOYEES = "SAVE_EMPLOYEES";
export const FETCH_EMPLOYEES = "FETCH_EMPLOYEES";

export const saveEmployees = (employees) => {
    return {
        type: SAVE_EMPLOYEES,
        payload: {
            employeeList: employees
        }
    };
};

export const fetchemployees = () => {
    return {
        type: FETCH_EMPLOYEES
    }
}
