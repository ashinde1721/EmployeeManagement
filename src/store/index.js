import { createStore, applyMiddleware } from "redux";
import rootReducer from "./reducer";
import createSagaMiddleware from 'redux-saga';

import {getemployeeSaga} from  '../sagas/employeesSaga'

const sagaMiddleware = createSagaMiddleware();

const store = createStore(rootReducer, applyMiddleware(sagaMiddleware));
sagaMiddleware.run(getemployeeSaga);

export default store;