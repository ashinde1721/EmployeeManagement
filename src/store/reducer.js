import * as actions from '../actions/actions';

const initialState = {
    search: '',
    employeeList: []
};

const reducer = (state = initialState, action) => {
    const { type, payload = {} } = action;

    switch (type) {
        case actions.SEARCH_TEXT:
            return { ...state, search: payload.search };

        case actions.SAVE_EMPLOYEES:
            return {
                ...state,
                employeeList: payload.employeeList
            }
        default:
            return state;
    }
};

export default reducer;
