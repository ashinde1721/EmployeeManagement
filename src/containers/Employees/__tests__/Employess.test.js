import React from 'react';
import { Provider } from 'react-redux';
import axios from 'axios';
import { cleanup, render, waitForElement } from 'react-testing-library';

import store from '../../../store';
import employees from '../Employees/Employees';
import { API_URL, API_KEY } from '../../../config/index'
import { SEARCH_TEXT } from '../../../actions/actions';


describe('employees Component', () => {
  let renderedComponent;
  const fetchemployees = async () => {
    const period = 7;
    const url = `${API_URL}/${period}.json?api-key=${API_KEY}`;
    const { data: { results = [] } } = await axios.get(url);
    return results;
  };
  
  beforeEach(() => {
    renderedComponent = render(
      <Provider store={store}>
        <employees />
      </Provider>
    );
  });

  afterEach(() => cleanup());

  it('renders component', () => {
    expect(renderedComponent.container).toMatchSnapshot();
  });

  it('should render all NY-Times-employees', async () => {
    const { getAllByTestId } = renderedComponent;

    const results = await fetchemployees();
    await waitForElement(() => getAllByTestId('nytimes-card'));
    expect(getAllByTestId('nytimes-card').length).toEqual(results.length);
  });

  it('should render only matched searchText NY-Times-employees', async () => {
    const searchText = 'The';
    store.dispatch({
      type: SEARCH_TEXT,
      payload: {
        search: searchText,
    }
    });
    const { getAllByTestId } = renderedComponent;

    const results = await fetchemployees();
    const matchedResults = results.filter(({ title }) => {
      if (title.toLowerCase().includes(searchText.toLowerCase())) {
        return true;
      }
      return false;
    });
    await waitForElement(() => getAllByTestId('nytimes-card'));
    expect(getAllByTestId('nytimes-card').length).toEqual(matchedResults.length);
  });
});
