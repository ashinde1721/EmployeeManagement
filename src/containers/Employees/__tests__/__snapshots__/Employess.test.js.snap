// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`employees Component renders component 1`] = `
<div>
  <employees />
</div>
`;
